rfcs/convergence/apperance-migration.md

```diff
@@ -0,0 +1,106 @@
+👎 Cons of adding appearance to achieve the same design as V0:
+
+- Not scalable, partners would have to add the prop to every component
+- If they decide to revert back to default appearance in future, they would have to go through the same pain again by removing the prop
+- Is error prone
+
+## Detailed Design or Proposal
+
+### Composed component
+
+Partners could create a new composed component in fluent and modify the props in their preferred way. If the partner would like to keep the original color, they could create the composed component and have the default component without the appearance prop renders as `filled-darker`.
+
+👍 Pros:
+
+- Is relatively safe
+
+👎 Cons:
+
+- Creating new composed component for each input `Dropdown`, `Input`, `TextArea` and `DatePicker`
+- Composed component apperance prop wouldn't match our Fluent V9 documentation
+- Does not work if repos have dependencies of another project with Fluent V9 input components (Nova)
+- Composition approach is currently marked as unstable in v9
+
+### Wrap the library components on application side
+
+Partners could a new component wrapper that will wrap input components and modify the appearance prop on the application side.
+
+👍 Pros:
+
+- Is safe and stable
+
+👎 Cons:
+
+- Creating new composed component for each input `Dropdown`, `Input`, `TextArea` and `DatePicker`
+- Wrapper component apperance wouldn't match our Fluent V9 documentation
+- Would work in iframes only within the TMP repository
+- Does not work if repos have dependencies of another project with Fluent V9 input components (Nova)
+
+### Global css selector
+
+Targeting all input selectors from a partner app and change the color with global css.
+
+Example: https://codesandbox.io/s/rfc-inputs-ew821q?file=/app.tsx
+
+👍 Pros:
+
+- Relatively easy and fast to do
+- Scalable
+- Would work for dependencies such as Nova (but depending where we inject the CSS rule and if they have their own `FluentProvider`)
+
+👎 Cons:
+
+- Difficult to validate
+
+### New token alias to theme
+
+Adding a new alias color token (let's say `colorInputBackground`) and use it for all inputs.
+
+👍 Pros:
+
+- Easy to do and would work for all inputs
+
+👎 Cons:
+
+- Negative impact on performance by increasing variables (as read here: [fluentui/theme-shared-colors.md at d5d510bf1ffcc1a4ed2067e9eb009c84e7beb351 · microsoft/fluentui (github.com)](https://github.com/microsoft/fluentui/blob/d5d510bf1ffcc1a4ed2067e9eb009c84e7beb351/rfcs/react-components/convergence/theme-shared-colors.md))
+- Divergence themes from the original
+
+Another option is to add a possibility to override the background using a CSS variable:
+We can use `backgroundColor: var(--inputBackgroundOverride, ${tokens.colorNeutralBackground1})` without setting the `--inputBackgroundOverride` anywhere. Then an application can set that variable if it needs to override the background.
+
+👍 no additional tokens unless needed
+
+👎 new concept
+👎 one-off just for the Input background
```

Comment:
But it's not only for `Input`, right? It's also for `Dropdown`, `TextArea` and other upcoming components, correct?

Reply:
Correct - there are multiple components affected but just one aspect is being tweaked right now. (single token?)